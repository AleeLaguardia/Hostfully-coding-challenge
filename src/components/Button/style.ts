import styled from "styled-components";

export const Container = styled.button<{
  color?: string;
  colorHover?: string;
  labelColor?: string;
  width?: string;
  borderColor?: string;
}>`
  display: flex;
  height: 40px;
  width: fit-content;
  align-items: center;
  justify-content: center;
  padding: ${(props) => props.width ? '' : '7px 30px'};
  width: ${(props) => props.width || ''}; 
  background-color: ${(props) => props.color};
  border-radius: 5px;
  border: 2px solid ${(props) => props.borderColor ? props.borderColor : props.color};
  cursor: pointer;

  &:hover {
    background-color: ${(props) => props.colorHover};
    border: 2px solid ${(props) => props.borderColor || props.colorHover};
  }

  span {
    font-family: ${(props) => props.theme.fonts.poppins};
    font-weight: 500;
    color: ${(props) => props.labelColor};
    font-size: ${(props) => props.theme.sizes.s};
  }
`;