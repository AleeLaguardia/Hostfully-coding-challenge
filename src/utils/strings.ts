export const PATH = {
  HOME: '/',
  BOOKING: '/booking',
  RESERVATION: '/reservation',
};

export const HOME = {
  TITLE_FORM: "Book your stay right now!",
  TITLE_PIC: "Book Your Stay",
  ADULTS: "How many adults",
  CHILDREN: "How many children",
}

export const BOOKING = {
  NO_DATA: "No hotels found in that area",
  ALSO_TRY: "But you can also try:"
};