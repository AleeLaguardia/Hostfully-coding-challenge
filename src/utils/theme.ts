import { DefaultTheme } from "styled-components";

import {
  white,
  alabaster,
  emperor,
  galery,
  gray,
  black,
  purple,
  purpleHover,
  darkPurple,
  radicalRed,
  lightPurple,
  silverChalice,
  alto,
  lynch,
  selectiveYellow,
  chelseaGem,
} from './styling/colors';
import {
  poppins,
  playwrite,
} from './styling/fonts';
import {
  xs,
  xxs,
  xxxs,
  s,
  m,
  l,
  xl,
  xxxl,
} from './styling/sizes';

export const theme: DefaultTheme = {
  colors: {
    white,
    alabaster,
    emperor,
    galery,
    gray,
    black,
    purple,
    purpleHover,
    darkPurple,
    radicalRed,
    lightPurple,
    silverChalice,
    alto,
    lynch,
    selectiveYellow,
    chelseaGem,
  },
  fonts: {
    poppins,
    playwrite,
  },
  sizes: {
    xs,
    xxs,
    xxxs,
    s,
    m,
    l,
    xl,
    xxxl,
  },
}