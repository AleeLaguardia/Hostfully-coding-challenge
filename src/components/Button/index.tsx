import React from "react";
import { Container } from "./style";
import { theme } from "../../utils/theme";

interface Props {
  label: string;
  onClick: () => void;
  color?: string;
  colorHover?: string;
  secondaryColor?: string;
  secondaryColorHover?: string;
  labelColor?: string;
  width?: string;
  borderColor?: string;
}

const Button: React.FC<Props> = ({
  label,
  onClick,
  color = theme.colors.purple,
  colorHover = theme.colors.purpleHover,
  secondaryColor = theme.colors.white,
  secondaryColorHover = theme.colors.alabaster,
  labelColor = theme.colors.white,
  width = '',
  borderColor,
}) => {
  return (
    <Container
      onClick={onClick}
      color={color}
      colorHover={colorHover}
      labelColor={labelColor}
      width={width}
      borderColor={borderColor}
    >
      <span>{label}</span>
    </Container>
  );
};

export default Button;