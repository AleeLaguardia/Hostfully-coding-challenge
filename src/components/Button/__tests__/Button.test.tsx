import React from 'react';
import { fireEvent, getByText, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '..';
import { theme } from '../../../utils/theme';
import { ThemeProvider } from 'styled-components';

describe('Button Component', () => {
  it('renders Button with the default style', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Button onClick={() => {}} label='test' type='primary' />
      </ThemeProvider>
    );

    expect(asFragment).toMatchSnapshot();
  });

  it('fires an event when clicked', () => {
    const setStateMock = jest.fn();

    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button onClick={() => setStateMock()} label='test' type='primary' />
      </ThemeProvider>
    );

    const button = getByText(container, 'test');
    fireEvent.click(button);
    expect(setStateMock).toHaveBeenCalled();
  })
});
