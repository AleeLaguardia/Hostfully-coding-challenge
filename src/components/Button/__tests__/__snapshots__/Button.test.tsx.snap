// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders Button with the default style 1`] = `[Function]`;
